app/guard/guard.go

```diff
@@ -0,0 +1,343 @@
+	GuHash := abstract.Sum(network.Suite, uid, epoch)
+	// creating stream for Scalar.Pick from the hash
+	// creating stream for Scalar.Pick from the hash
+	blocky, err := aes.NewCipher(iv)
+	log.ErrFatal(err)
+	log.Lvl2("Blocky: ", blocky)
+	log.Lvl2("Iv: ", iv)
+	log.Lvl2("GetHash: ", pwhash)
+	GuStream := cipher.NewCTR(blocky, iv)
+	gupoint := network.Suite.Point()
+	Gu, _ := gupoint.Pick(GuHash, GuStream)
+	// printing gudat is just a test
+	Gudat, _ := Gu.MarshalBinary()
+	log.Lvl2("GetGu: ", Gudat)
+
+	for i, si := range db.Cothority.List {
+		cl := guard.NewClient()
+		// blankpoints needed for computations
+		blankpoint := network.Suite.Point()
+		blankscalar := network.Suite.Scalar()
+		// pwbytes and blindbytes are actually scalars that are initialized to the values of the bytes
+		pwbytes := network.Suite.Scalar()
+		pwbytes.SetBytes(pwhash)
+		a, _ := blankscalar.MarshalBinary()
+		log.Lvl2("BlankScalar: ", a)
+		blindbytes := network.Suite.Scalar()
+		blindbytes.SetBytes(blinds[i])
+		b, _ := blankscalar.MarshalBinary()
+		// The following sections of the code perform the computations to Create Xi, here called sendy.
+		log.Lvl2("BlankScalar: ", b)
+		log.Lvl2("getpwhash: ", pwhash)
+		log.Lvl2("getblind: ", blinds[i])
+		ad, _ := blankscalar.Add(pwbytes, blindbytes).MarshalBinary()
+		log.Lvl2("GetAddition: ", ad)
+		mul, err := blankpoint.Mul(Gu, blankscalar.Mul(pwbytes, blindbytes)).MarshalBinary()
+		log.Lvl2("GetMul: ", mul)
+		sendy := blankpoint.Mul(Gu, blankscalar.Mul(pwbytes, blindbytes))
+		rep, err := cl.GetGuard(si, uid, epoch, sendy)
+		log.Lvl2("GetRep: ", rep.Msg)
+		log.ErrFatal(err)
+		// This section of the program removes the blinding factor from the Zi for storage
+		reply, err := blankpoint.Mul(rep.Msg, blankscalar.Inv(blindbytes)).MarshalBinary()
+		log.ErrFatal(err)
+		log.Lvl2("Reply: ", reply)
+		// This section Xors the data with the response
+		block, err := aes.NewCipher(reply)
+		stream := cipher.NewCTR(block, getuser(uid).Iv)
+		msg := make([]byte, len([]byte(getuser(uid).Keys[i])))
+		stream.XORKeyStream(msg, []byte(getuser(uid).Keys[i]))
+		keys[i] = string(msg)
+	}
+	k := s.Combine(keys)
+	if len(k) == 0 {
+		log.Fatal("You entered the wrong password")
+	}
+	log.Lvl2("key: ", []byte(k))
+	block, err := aes.NewCipher([]byte(k))
+	log.ErrFatal(err)
+	aesgcm, err := cipher.NewGCM(block)
+	log.ErrFatal(err)
+	plaintext, err := aesgcm.Open(nil, getuser(uid).Salt, getuser(uid).Data, nil)
+	log.ErrFatal(err)
+	log.Print(string(plaintext))
+}
+func setpass(c *cli.Context) error {
+	UID := []byte(c.Args().Get(0))
+	Epoch := []byte{'E', 'P', 'O', 'C', 'H'}
+	Pass := c.Args().Get(1)
+	usrdata := []byte(c.Args().Get(2))
+	set(c, UID, Epoch, string(Pass), usrdata)
+	b, err := network.MarshalRegisteredType(db)
+	log.ErrFatal(err)
+	err = ioutil.WriteFile("config.bin", b, 0660)
+	return nil
+}
+func get(c *cli.Context) error {
+	uid := []byte(c.Args().Get(0))
+	epoch := []byte{'E', 'P', 'O', 'C', 'H'}
```

Comment:
In case you struggle with the above proposal: I thinl @ineiti meant `[]byte("EPOCH")` instead.